securesystemslib/hsm.py

```diff
@@ -0,0 +1,446 @@
+#!/usr/bin/env python
+
+"""
+<Program Name>
+  hsm.py
+
+<Author>
+  Tanishq Jasoria <jasoriatanishq@gmail.com>
+
+<Purpose>
+  The goal of this module is to support hardware security modules through
+  the PKCS#11 standard.
+
+  This module uses PyKCS11, a python wrapper (SWIG) for PKCS#11 modules
+  to communicate with the cryptographic tokens
+"""
+
+# Help with Python 3 compatibility, where the print statement is a function, an
+# implicit relative import is invalid, and the '/' operator performs true
+# division.  Example:  print 'hello world' raises a 'SyntaxError' exception.
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+# Import python wrapper for PKCS#11 to communicate with the tokens
+import PyKCS11
+
+import binascii
+import logging
+import securesystemslib.exceptions
+
+# Import cryptography routines needed to retrieve cryptographic
+# keys and certificates in PEM format.
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.backends import default_backend
+from cryptography import x509
+
+# RSA-PSS with SHA256 hash to be used for signature generation.
+RSA_PSS_MECH = PyKCS11.CKM_SHA256_RSA_PKCS_PSS
+# SHA256 hash to be used to digest the data.
+RSA_PSS_HASH_SHA256 = PyKCS11.CKM_SHA256
+# Mask generating function for SHA256 Hash.
+RSA_PSS_MGF_SHA256 = PyKCS11.CKG_MGF1_SHA256
+# Length of salt to be used for hashing.
+RSA_PSS_SALT_LENGTH = 32
+
+logger = logging.getLogger('securesystemslib_hsm')
```

Comment:
Mostly a nitpick, but `getLogger(__name__)` is a common practice for initializing module-level loggers. `__name__` also makes it easier to enable or disable particular log-levels in client code that consumes this library.